packages/agents/src/agents/Chameleon/ContextualizedChat.ts

```diff
@@ -0,0 +1,294 @@
+import {
```

Comment:
General feedback on this file:
While tracking chunks is efficient, it also seems to make the whole implementation rather unwieldly
I'd recommend either refactoring this file to make it more easier to understand 
or (which might be simpler to implement and read)
Keeping an instance of raw chat without chunks, and then only chunking when it's immediately needed

Reply:
Would you mind being more specific in regards to what parts of the implementation are unwieldly?

I took a fresh read through this file line-by-line this morning and feel pretty good about its readability and maintainability. I think it abides by the "single responsibility" principle well, and the code flow reads well in plain english making it easy to understand.

I'm of course biased, which is why I'm seeking specific feedback on what parts are confusing to you.

Reply:
> Keeping an instance of raw chat without chunks
This is being done, the raw chat reference is added in the constructor (`this._chat`)

> only chunking when it's immediately needed
This is being done, processing new messages is done when `contextualize()` is called. Only large messages are chunked.